stdlib/test/shutil/test_shutil.mojo

```diff
@@ -0,0 +1,54 @@
+# ===----------------------------------------------------------------------=== #
+# Copyright (c) 2024, Modular Inc. All rights reserved.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions:
+# https://llvm.org/LICENSE.txt
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===----------------------------------------------------------------------=== #
+# RUN: %mojo %s
+
+from shutil import rmtree
+from os import mkdir
+from os.path import exists
+from pathlib import Path
+
+from testing import assert_false, assert_raises
+
+
+fn _create_some_files_and_dirs(path: Path) raises:
+    mkdir(path / "nested")
+    mkdir(path / "nested" / "more_nested")
+    mkdir(path / "nested" / "more_nested" / "even_more_nested")
+
+    _ = open(path / "nested.txt", "w")
+    _ = open(path / "nested" / "more_nested.txt", "w")
+    _ = open(path / "nested" / "more_nested" / "even_more_nested.txt", "w")
+
+
+fn test_rmtree() raises:
+    var cwd_path = Path()
+    var my_dir_path = cwd_path / "my_dir"
+
+    assert_false(
+        my_dir_path.exists(),
+        "Unexpected dir " + my_dir_path.__fspath__() + " it should not exist",
+    )
+
+    with assert_raises():
+        rmtree(my_dir_path)
+
+    mkdir(my_dir_path)
+    _create_some_files_and_dirs(my_dir_path)
+
+    rmtree(my_dir_path)
```

Comment:
Testing these kinds of functions can actually be tricky: what if the test fails for some reason and the directory isn't deleted? The next time the test is run, `mkdir` can fail. One way to fix this is by doing this in a context manager, like [`tempfile.TemporaryDirectory`](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory). In lieu of that, some manual cleanup logic could also work.

Reply:
yes, it's tricky, but doing some manual cleanup is not that simple in this case, what if what is failing is `rmdir/unlink` ? obviously `rmtree` is relying on those functions so what else could be done?

we have the same "issue" in `test_remove` and `test_mkdir_and_rmdir`